server_extensions_extra/src/main/java/com/generalbytes/batm/server/extensions/extra/bitcoin/exchanges/bittrex/BittrexExchange.java

```diff
@@ -303,119 +294,6 @@ public String getDepositAddress(String cryptoCurrency) {
         return null;
     }
 
-    @Override
-    public BigDecimal getExchangeRateForBuy(String cryptoCurrency, String fiatCurrency) {
-        BigDecimal result = calculateBuyPrice(cryptoCurrency, fiatCurrency, getMeasureCryptoAmount(cryptoCurrency));
-        if (result != null) {
-            return result.divide(getMeasureCryptoAmount(cryptoCurrency), 2, BigDecimal.ROUND_UP);
-        }
-        return null;
-    }
-
-    @Override
-    public BigDecimal getExchangeRateForSell(String cryptoCurrency, String fiatCurrency) {
-        BigDecimal result = calculateSellPrice(cryptoCurrency, fiatCurrency, getMeasureCryptoAmount(cryptoCurrency));
-        if (result != null) {
-            return result.divide(getMeasureCryptoAmount(cryptoCurrency), 2, BigDecimal.ROUND_DOWN);
-        }
-        return null;
-    }
-
-    private BigDecimal getMeasureCryptoAmount(String cryptoCurrency) {
-        if (CryptoCurrency.BTC.getCode().equals(cryptoCurrency)) {
-            return XChangeExchange.BTC_RATE_SOURCE_CRYPTO_AMOUNT;
-        }
-        return new BigDecimal(5);
-    }
-
-    @Override
-    public BigDecimal calculateBuyPrice(String cryptoCurrency, String fiatCurrency, BigDecimal cryptoAmount) {
-        try {
-            MarketDataService marketDataService = getExchange().getMarketDataService();
-            RateLimiter.waitForPossibleCall(getClass());
-
-            CurrencyPair currencyPair = new CurrencyPair(cryptoCurrency, fiatCurrency);
-            List<LimitOrder> asks = marketDataService.getOrderBook(currencyPair).getAsks();
-            Collections.sort(asks, asksComparator);
-
-            BigDecimal tradableLimit = getTradablePrice(cryptoAmount, asks);
```

Comment:
the point of this complicated way of getting the price is that if you want to buy a high volume of some asset on the exchange you have to pay higher price than for a small volume. E.g. if you're buying you would accept (buy) all the cheap offers on the exchange (order book) and then you also accept the offers for higher and higher prices until you have the volume you wanted to buy.
With selling it would be the same except the price for selling a large volume goes down instead.

Would it be possibe to keep this in the new implementation? (including the getMeasureCryptoAmount method for when the amount is unknown)

Reply:
Thanks for explanation. I left it unchanged in the new implementation.